globset/src/glob.rs

```diff
@@ -194,6 +194,8 @@ struct GlobOptions {
     /// Whether to require a literal separator to match a separator in a file
     /// path. e.g., when enabled, `*` won't match `/`.
     literal_separator: bool,
+    /// doc
+    from_regex: bool,
```

Comment:
I'm not sure this is quite the right way to go about this. In particular, you wind up storing a regex in a field called `glob`, which doesn't seem good to me. Additionally, the `from_regex` option isn't really compatible with the `case_insensitive` or `literal_separator` flags, which also makes the builder strange. I think I'd rather see `Glob` itself be an enum and support a `from_regex` constructor. In that approach, `GlobBuilder` doesn't need to be modified at all.